packages/babel-plugin-transform-modules-amd/src/index.ts

```diff
@@ -12,33 +13,35 @@ import {
 import { template, types as t } from "@babel/core";
 import { getImportSource } from "babel-plugin-dynamic-import-node/utils";
 import type { PluginOptions } from "@babel/helper-module-transforms";
+import type { NodePath } from "@babel/traverse";
 
-const buildWrapper = template(`
+const buildWrapper = template.statement(`
   define(MODULE_NAME, AMD_ARGUMENTS, function(IMPORT_NAMES) {
   })
 `);
 
-const buildAnonymousWrapper = template(`
+const buildAnonymousWrapper = template.statement(`
   define(["require"], function(REQUIRE) {
   })
 `);
 
-function injectWrapper(path, wrapper) {
+function injectWrapper(path: NodePath<t.Program>, wrapper: t.Statement) {
   const { body, directives } = path.node;
   path.node.directives = [];
   path.node.body = [];
   const amdWrapper = path.pushContainer("body", wrapper)[0];
-  const amdFactory = amdWrapper
-    .get("expression.arguments")
+  const amdFactory = (
+    amdWrapper.get("expression.arguments") as NodePath<t.Expression>[]
+  )
     .filter(arg => arg.isFunctionExpression())[0]
```

Comment:
Nit (not related to this PR, but we can fix it here since it's small): `.find()` instead of `.filter()[0]`.